test/framework/resources/aws/services/eks.go

```diff
@@ -47,12 +47,20 @@ type AddonInput struct {
 
 func NewEKS(cfg aws.Config, endpoint string) (EKS, error) {
 	var err error
-
+	var customResolver aws.EndpointResolverWithOptions
 	if endpoint != "" {
-		customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
-			return aws.Endpoint{
-				URL: endpoint,
-			}, nil
+		// EKS Custom endpoint resolver needs PartitionID, SingingRegion and URL for handling STS requests.
+		// TODO: default to "aws" partition for now as it handled only tests. Provide option to pass partitionID.
+		customResolver = aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
+			if service == eks.ServiceID {
+				return aws.Endpoint{
+					PartitionID:   "aws",
```

Comment:
Will this work for regions in `aws-cn/aws-gov` partition?

Reply:
Good point. I will have to test. I will see how we are sending the endpoint in those partitions.

Reply:
If we are not sending a custom `--eks-endpoint` (that is a custom endpoint cluster in the particular partion, we should be good), otherwise partition id needs to sent separately.

Reply:
Chatted offline; endpoint is only set in beta tests so this is good to go.